pagai/services/postgres/main.py

```diff
@@ -122,28 +135,45 @@ def get_table_names(connection=None):
 
     for partition in partitions:
         partition_table_names = partition[0].split(", ")
-        all_table_names = list(
-            filter(lambda x: x[0] not in partition_table_names, all_table_names)
-        )
+        all_table_names = list(filter(lambda x: x[0] not in partition_table_names, all_table_names))
 
     tables = np.array(all_table_names).T[0]
 
     return tables
 
 
 @cache
-def get_table(table, connection=None, limit=1000):
+def explore(table, connection, limit, schema=None):
+    """
+    Returns the first rows of a table alongside the column names.
+    """
+    try:
+        return {
+            "fields": get_column_names(table, schema=schema, connection=connection).tolist(),
+            "rows": get_table(table, schema=schema, connection=connection, limit=limit).tolist(),
+        }
+    except psycopg2.OperationalError as e:
+        raise OperationOutcome(e)
+    except psycopg2.errors.UndefinedTable:
+        raise OperationOutcome(f"Table {table} does not exist in database")
+
+
+@cache
+def get_table(table, connection=None, limit=1000, schema=None):
```

Comment:
Do we want an argument to choose the sampling?
If understood well, rows are chosen at random now but maybe it could be useful to have the first ones, last ones, etc.

Reply:
yes it's a good point but not in the scope of this PR, if we need this I'll add it later

Reply:
Right!